File: cpp/program/Barang.cpp
#include <iostream>

using namespace std;

/* Class untuk menyimpan data barang di toko Semua data barang 
disimpan dalam variabel private supaya aman */
class Barang{
    protected:
        // Data-data barang yang disimpan sebagai private
        // Kenapa private? Supaya tidak bisa diubah sembarangan dari luar class
        string kode; // Kode unik barang
        string nama;  // Nama barang
        string harga; // Harga barang 
        string merk; // Merk barang
        string garansi;  // Lama garansi
        string stok; // Jumlah barang yang tersedia

    public:
        // Constructor kosong - bikin objek barang dengan data kosong
        // Dipake kalau mau bikin array barang tapi belum tau isinya, karena datanya di isi oleh user
        Barang() {
            kode = "";
            nama = "";
            harga = "";
            merk = "";
            garansi = "";
            stok = "";
        }

        // Constructor lengkap - langsung isi semua data barang sekaligus
        // Dipake kalau udah tau semua data barangnya
        Barang(string kode, string nama, string harga, string merk, string garansi, string stok){
            this->kode = kode;
            this->nama = nama;
            this->harga = harga;
            this->merk = merk;
            this->garansi = garansi;
            this->stok = stok;
        }

        // Getter dan Setter - cara aman untuk akses data private
        // Getter untuk ambil data, Setter untuk ubah data
        string getKode(){
            return kode;
        }

        void setKode(string kode){
            this->kode = kode;
        }

        string getNama(){
            return nama;
        }

        void setNama(string nama){
            this->nama = nama;
        }

        string getHarga(){
            return harga;
        }

        void setHarga(string harga){
            this->harga = harga;
        }

        string getMerk(){
            return merk;
        }

        void setMerk(string merk){
            this->merk = merk;
        }

        string getGaransi(){
            return garansi;
        }

        void setGaransi(string garansi){
            this->garansi = garansi;
        }

        string getStok(){
            return stok;
        }

        void setStok(string stok){
            this->stok = stok;
        }

};

File: php/program/Barang.php
<?php
class Barang{
    // Properti untuk menyimpan data barang
    private string $kode;
    private string $nama;
    private string $harga;
    private string $merk;
    private string $garansi;
    private string $stok;
    private string $gambar;

    // Konstruktor untuk inisialisasi data barang saat objek dibuat
    public function __construct(string $kode, string $nama, string $harga, string $merk, string $garansi, string $stok, string $gambar){
        $this->kode = $kode;
        $this->nama = $nama;
        $this->harga = $harga;
        $this->merk = $merk;
        $this->garansi = $garansi;
        $this->stok = $stok;
        $this->gambar = $gambar;
    }

    // Getter dan Setter untuk kode
    public function getKode(): string {
        return $this->kode;
    }
    public function setKode(string $kode): void{
        $this->kode = $kode;
    }

    // Getter dan Setter untuk nama
    public function getNama(): string {
        return $this->nama;
    }
    public function setNama(string $nama): void{
        $this->nama = $nama;
    }

    // Getter dan Setter untuk harga
    public function getHarga(): string {
        return $this->harga;
    }
    public function setHarga(string $harga): void{
        $this->harga = $harga;
    }

    // Getter dan Setter untuk merk
    public function getMerk(): string {
        return $this->merk;
    }
    public function setMerk(string $merk): void{
        $this->merk = $merk;
    }

    // Getter dan Setter untuk garansi
    public function getGaransi(): string {
        return $this->garansi;
    }
    public function setGaransi(string $garansi): void{
        $this->garansi = $garansi;
    }

    // Getter dan Setter untuk stok
    public function getStok(): string {
        return $this->stok;
    }
    public function setStok(string $stok): void{
        $this->stok = $stok;
    }

    // Getter dan Setter untuk gambar
    public function getGambar(): string {
        return $this->gambar;
    }
    public function setGambar(string $gambar): void {
        $this->gambar = $gambar;
    }
}
?>
